build-llvm.py

```diff
@@ -0,0 +1,402 @@
+    # The projects to build
+    defines['LLVM_ENABLE_PROJECTS'] = projects
+    # Don't build bindings; they are for other languages that the kernel does not use
+    defines['LLVM_ENABLE_BINDINGS'] = 'OFF'
+    # Don't build Ocaml documentation
+    defines['LLVM_ENABLE_OCAMLDOC'] = 'OFF'
+    # Removes system dependency on terminfo and almost every major clang provider turns this off
+    defines['LLVM_ENABLE_TERMINFO'] = 'OFF'
+    # Don't build clang-tools-extras to cut down on build targets (about 400 files or so)
+    defines['LLVM_EXTERNAL_CLANG_TOOLS_EXTRA_SOURCE_DIR'] = ''
+    # Don't include documentation build targets because it is available on the web
+    defines['LLVM_INCLUDE_DOCS'] = ''
+    # Don't include example build targets to save on cmake cycles
+    defines['LLVM_INCLUDE_EXAMPLES'] = 'OFF'
+    # The architectures to build backends for
+    defines['LLVM_TARGETS_TO_BUILD'] = targets
+
+    # If a debug build was requested
+    if debug:
+        defines['CMAKE_BUILD_TYPE'] = 'Debug'
+        defines['CMAKE_C_FLAGS'] = '-march=native -mtune=native'
+        defines['CMAKE_CXX_FLAGS'] = '-march=native -mtune=native'
+        defines['LLVM_BUILD_TESTS'] = 'ON'
+    # If a release build was requested
+    else:
+        defines['CMAKE_BUILD_TYPE'] = 'Release'
+        defines['CMAKE_C_FLAGS'] = '-O2 -march=native -mtune=native'
+        defines['CMAKE_CXX_FLAGS'] = '-O2 -march=native -mtune=native'
+        defines['LLVM_INCLUDE_TESTS'] = 'OFF'
+        defines['LLVM_ENABLE_WARNINGS'] = 'OFF'
+
+    # Don't build libfuzzer when compiler-rt is enabled, it invokes cmake again and we don't use it
+    if "compiler-rt" in projects:
+        defines['COMPILER_RT_BUILD_LIBFUZZER'] = 'OFF'
+
+    # Use ccache if it is available for faster incremental builds
+    if shutil.which("ccache") is not None:
+        defines['LLVM_CCACHE_BUILD'] = 'ON'
+
+    # If we found a linker, we should use it
+    if ld is not None:
+        defines['LLVM_USE_LINKER'] = ld
+
+    # Add the defines, point them to our build folder, and invoke cmake
+    cmake = ['cmake', '-G', 'Ninja', '-Wno-dev']
+    for key in defines:
+        newdef = '-D' + key + '=' + defines[key]
+        cmake += [newdef]
+    cmake += [root.joinpath("llvm-project", "llvm").as_posix()]
+
+    subprocess.run(cmake, check=True, cwd=build.as_posix())
+
+
+# Build the world
+def invoke_ninja(build, install_folder):
+    utils.header("Building LLVM")
+
+    timeStarted = time.time()
+
+    subprocess.run('ninja', check=True, cwd=build.as_posix())
+
+    print()
+    print("LLVM build duration: " +
+          str(datetime.timedelta(seconds=int(time.time() - timeStarted))))
+
+    subprocess.run(['ninja', 'install'],
+                   check=True,
+                   cwd=build.as_posix(),
+                   stdout=subprocess.DEVNULL,
+                   stderr=subprocess.DEVNULL)
+
+    utils.gitignore(install_folder)
```

Comment:
```suggestion
    utils.create_gitignore(install_folder)
```

Reply:
Done locally, thanks (that's what I get for doing last minute changes without actually running the script)